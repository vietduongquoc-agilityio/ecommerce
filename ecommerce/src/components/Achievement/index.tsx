"use client";

import { fonts, colors } from "@/themes";
import Image, { StaticImageData } from "next/image";
import Quality from "@/assets/Images/Shop-quality.png";
import Protection from "@/assets/Images/Shop-protection.png";
import Shipping from "@/assets/Images/Shop-shipping.png";
import Support from "@/assets/Images/Shop-support.png";

interface ItemCardProps {
  image: string | StaticImageData;
  alt: string;
  title: string;
  description: string;
}

const AchievementItem: React.FC<ItemCardProps> = ({ image, alt, title, description }) => (
  <li style={achievementItemStyle}>
    <Image src={image} alt={alt} />
    <div>
      <p style={achievementTitleStyle}>{title}</p>
      <p style={achievementDescriptionStyle}>{description}</p>
    </div>
  </li>
);

const achievementItemStyle = {
  display: "flex",
  gap: "14px",
  alignItems: "center",
};

const achievementTitleStyle = {
  fontSize: fonts.size.lg,
  fontWeight: fonts.weight.semiBold,
};

const achievementDescriptionStyle = {
  fontSize: fonts.size.md,
  fontWeight: fonts.weight.medium,
  color: colors.secondary.splight,
};

const Achievement = () => {
  return (
    <section
      style={{
        width: "100%",
        height: "auto",
        background: colors.secondary.linen,
        marginBottom: "-32px",
      }}
    >
      <ul
        style={{
          padding: "100px 53px",
          listStyle: "none",
          display: "flex",
          justifyContent: "space-between",
        }}
      >
        <AchievementItem
          image={Quality}
          alt={"quality"}
          title={"High Quality"}
          description={"crafted from top materials"}
        />
        <AchievementItem
          image={Protection}
          alt={"protection"}
          title={"Warranty Protection"}
          description={"Over 2 years"}
        />
        <AchievementItem
          image={Shipping}
          alt={"shipping"}
          title={"Free Shipping"}
          description={"Order over 150 $"}
        />
        <AchievementItem
          image={Support}
          alt={"support"}
          title={"24 / 7 Support"}
          description={"Dedicated support"}
        />
      </ul>
    </section>
  );
};

export default Achievement;
