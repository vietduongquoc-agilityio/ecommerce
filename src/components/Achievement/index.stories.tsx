import { Meta, StoryFn } from "@storybook/react";
import Achievement from "./";
import Quality from "@/assets/Images/Shop-quality.png";
import Protection from "@/assets/Images/Shop-protection.png";
import Shipping from "@/assets/Images/Shop-shipping.png";
import Support from "@/assets/Images/Shop-support.png";

export default {
  title: "Components/Achievement",
  component: Achievement,
  tags: ["autodocs"],
  args: {
    items: [
      {
        image: Quality,
        alt: "quality",
        title: "High Quality",
        description: "crafted from top materials",
      },
      {
        image: Protection,
        alt: "protection",
        title: "Warranty Protection",
        description: "Over 2 years",
      },
      {
        image: Shipping,
        alt: "shipping",
        title: "Free Shipping",
        description: "Order over 150 $",
      },
      {
        image: Support,
        alt: "support",
        title: "24 / 7 Support",
        description: "Dedicated support",
      },
    ],
  },
} as Meta;

const Template: StoryFn = (args) => <Achievement {...args} />;

export const Default = Template.bind({});
Default.args = {
  items: [
    {
      image: Quality,
      alt: "quality",
      title: "High Quality",
      description: "crafted from top materials",
    },
    {
      image: Protection,
      alt: "protection",
      title: "Warranty Protection",
      description: "Over 2 years",
    },
    {
      image: Shipping,
      alt: "shipping",
      title: "Free Shipping",
      description: "Order over 150 $",
    },
    {
      image: Support,
      alt: "support",
      title: "24 / 7 Support",
      description: "Dedicated support",
    },
  ],
};
