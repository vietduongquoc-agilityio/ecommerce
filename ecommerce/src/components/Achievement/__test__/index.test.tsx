import React from "react";
import { render } from "@testing-library/react";
import Achievement from "../";

describe("Achievement Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Achievement />);
    expect(asFragment()).toMatchSnapshot();
  });
});
