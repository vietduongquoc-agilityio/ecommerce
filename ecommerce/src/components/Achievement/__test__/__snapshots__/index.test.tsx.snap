// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Achievement Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    style="width: 100%; height: auto; background: rgb(250, 243, 234); margin-bottom: -32px;"
  >
    <ul
      style="padding: 100px 53px; list-style: none; display: flex; justify-content: space-between;"
    >
      <li
        style="display: flex; gap: 14px; align-items: center;"
      >
        <img
          alt="quality"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div>
          <p
            style="font-size: 24px; font-weight: 600;"
          >
            High Quality
          </p>
          <p
            style="font-size: 20px; font-weight: 500; color: rgb(159, 159, 159);"
          >
            crafted from top materials
          </p>
        </div>
      </li>
      <li
        style="display: flex; gap: 14px; align-items: center;"
      >
        <img
          alt="protection"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div>
          <p
            style="font-size: 24px; font-weight: 600;"
          >
            Warranty Protection
          </p>
          <p
            style="font-size: 20px; font-weight: 500; color: rgb(159, 159, 159);"
          >
            Over 2 years
          </p>
        </div>
      </li>
      <li
        style="display: flex; gap: 14px; align-items: center;"
      >
        <img
          alt="shipping"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div>
          <p
            style="font-size: 24px; font-weight: 600;"
          >
            Free Shipping
          </p>
          <p
            style="font-size: 20px; font-weight: 500; color: rgb(159, 159, 159);"
          >
            Order over 150 $
          </p>
        </div>
      </li>
      <li
        style="display: flex; gap: 14px; align-items: center;"
      >
        <img
          alt="support"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div>
          <p
            style="font-size: 24px; font-weight: 600;"
          >
            24 / 7 Support
          </p>
          <p
            style="font-size: 20px; font-weight: 500; color: rgb(159, 159, 159);"
          >
            Dedicated support
          </p>
        </div>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
